File: Src/Icm.ContextConsole/Context/ReflectionActionFinder.vb
''' <summary>
''' Finds actions by using reflection on the context.
''' </summary>
''' <remarks>It finds all public non-static parameterless procedures.</remarks>
Public Class ReflectionActionFinder
    Implements IActionFinder

    Public Function GetActions(ctl As IContext) As IEnumerable(Of IAction) Implements IActionFinder.GetActions
        Dim methods = ctl.GetType.GetMethods().Where(Function(minf) _
                                minf.ReturnType.Name = "Void" AndAlso _
                                minf.GetParameters().Count = 0 AndAlso _
                                minf.IsPublic AndAlso _
                                Not minf.IsStatic)

#If Framework = "net35" Then
        Return methods.Select(Function(minf) New MethodInfoAction(minf, ctl).As(Of IAction)())
#Else
        Return methods.Select(Function(minf) New MethodInfoAction(minf, ctl))
#End If
    End Function
End Class

File: Src/Icm.ContextConsole/Context/ReflectionActionFinder.cs
using System.Collections.Generic;
using System.Linq;

/// <summary>
/// Finds actions by using reflection on the context.
/// </summary>
/// <remarks>It finds all public non-static parameterless procedures.</remarks>
public class ReflectionActionFinder : IActionFinder
{

	public IEnumerable<IAction> GetActions(IContext ctl)
	{
		var methods = ctl.GetType().GetMethods().Where(minf => minf.ReturnType.Name == "Void" && !minf.GetParameters().Any() && minf.IsPublic && !minf.IsStatic);
        return methods.Select(minf => new MethodInfoAction(minf, ctl));
    }
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================